python3.spec

```diff
@@ -47,6 +47,7 @@ make %{makeprocesses} install
 sed -i -e "s|^#!.*python%{pythonv} *$|#!/usr/bin/env python%{python_major}|" %{i}/bin/* %{i}/lib/python*/*.py
 sed -i -e 's|^#!/.*|#!/usr/bin/env python%{pythonv}m|' %{i}/lib/python*/config-*/python-config.py
 sed -i -e 's|^#! */usr/local/bin/python|#!/usr/bin/env python|' %{i}/lib/python*/cgi.py
+sed -i -e 's|^ENABLE_USER_SITE = .*|ENABLE_USER_SITE = False|g' %{i}/lib/python*/site.py
```

Comment:
Actually we do not want this. This disables `PYTHONUSERBASE` env which we use to install pip packages.
We can recommend users to set `PYTHONUSERBASE` to some non valid path. This avoid adding `~/.local` in the pythonpath. see code from `site.py`
```
# Copy of sysconfig._getuserbase()
def _getuserbase():
    env_base = os.environ.get("PYTHONUSERBASE", None)
    if env_base:
        return env_base

    def joinuser(*args):
        return os.path.expanduser(os.path.join(*args))

    if os.name == "nt":
        base = os.environ.get("APPDATA") or "~"
        return joinuser(base, "Python")

    if sys.platform == "darwin" and sys._framework:
        return joinuser("~", "Library", sys._framework,
                        "%d.%d" % sys.version_info[:2])

    return joinuser("~", ".local")

```

Reply:
@smuzaffar does this imply that we should not recommend users to set `export PYTHONNOUSERSITE=True`?

What if we reverse the logic of this function? i.e., treat `PYTHONUSERBASE` as null by default and only consider it if the user deliberately set it to some value.

Reply:
humm, I did not know about `PYTHONNOUSERSITE`. So is `export PYTHONNOUSERSITE=True`  not enough to avoid adding `~/.local` to PYTHONPATH?

Reply:
I think yes: https://docs.python.org/3.3/library/site.html#site.ENABLE_USER_SITE

Reply:
https://docs.python.org/3.3/using/cmdline.html#envvar-PYTHONNOUSERSITE

Reply:
Instead of changing python behavior (which can break users code who were using user site customization or site-dir`), I think we should recommend users to set `PYTHONNOUSERSITE=True` to avoid local modules from `~/.lcoal`

- without PYTHONNOUSERSITE=True
```
bash-5.1$ python3 -c 'import site;print(site.getusersitepackages());import muzaffar;print(muzaffar.MYNAME)'
/afs/cern.ch/user/m/muzaffar/.local/lib/python3.9/site-packages
muzaffar
```
- with PYTHONNOUSERSITE=True
```
bash-5.1$ PYTHONNOUSERSITE=True python3 -c 'import site;print(site.getusersitepackages());import muzaffar;print(muzaffar.MYNAME)'
/afs/cern.ch/user/m/muzaffar/.local/lib/python3.9/site-packages
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'muzaffar'
```